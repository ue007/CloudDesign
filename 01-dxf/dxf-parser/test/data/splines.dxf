  0
SECTION
  2
HEADER
  0
ENDSEC
  0
SECTION
  2
ENTITIES
  0
SPLINE
  5
3D
330
157
100
AcDbEntity
  8
PLOTLINES
100
AcDbSpline
210
0.0
220
0.0
230
1.0
 70
  1064
 71
     3
 72
     9
 73
     5
 74
     3
 42
0.000000001
 43
0.0000000001
 44
0.0
 12
0.9514183274466854
 22
0.3079012279913022
 32
0.0
 40
0.0
 40
0.0
 40
0.0
 40
0.0
 40
1547.322808138777
 40
5019.938770381674
 40
5019.938770381674
 40
5019.938770381674
 40
5019.938770381674
 10
0.0
 20
0.0
 30
0.0
 10
490.717092713168
 20
158.8075309082932
 30
0.0
 10
2164.905820751131
 20
388.1223313710202
 30
0.0
 10
3837.689038507039
 20
353.4902462273104
 30
0.0
 10
4994.861260930288
 20
329.5329955163307
 30
0.0
 11
0.0
 21
0.0
 31
0.0
 11
1522.67200320285
 21
275.0960618560202
 31
0.0
 11
4994.861260930288
 21
329.5329955163307
 31
0.0
  0
SPLINE
  5
3E
330
157
100
AcDbEntity
  8
PLOTLINES
100
AcDbSpline
210
0.0
220
0.0
230
1.0
 70
     8
 71
     3
 72
    26
 73
    22
 74
     0
 42
0.000000001
 43
0.001
 40
0.0
 40
0.0
 40
0.0
 40
0.0
 40
294.0525150032295
 40
294.0525150032295
 40
1045.910856597887
 40
1045.910856597887
 40
1224.445418796694
 40
1224.445418796694
 40
1454.016505907184
 40
1454.016505907184
 40
1572.806679130005
 40
1572.806679130005
 40
1691.596852352826
 40
1691.596852352826
 40
1929.177198798468
 40
1929.177198798468
 40
2404.337891689753
 40
2404.337891689753
 40
3354.659277472323
 40
3354.659277472323
 40
5015.653135237717
 40
5015.653135237717
 40
5015.653135237717
 40
5015.653135237717
 10
-3.910345595490071
 20
12.08301275857048
 30
0.0
 10
89.61730010503788
 20
42.35074686835108
 30
0.0
 10
184.0439740094698
 20
69.42067550989128
 30
0.0
 10
522.4770607652581
 20
155.8817672278711
 30
0.0
 10
770.4801029905291
 20
199.9804308248713
 30
0.0
 10
1079.587521541959
 20
241.21623757007
 30
0.0
 10
1139.003629790081
 20
248.5213454592013
 30
0.0
 10
1274.962277830939
 20
264.0530430007511
 30
0.0
 10
1351.545278854538
 20
271.9460988128703
 30
0.0
 10
1467.784975350118
 20
282.9561993903517
 30
0.0
 10
1507.426369672711
 20
286.5407381210116
 30
0.0
 10
1586.7020874704
 20
293.4221664132911
 30
0.0
 10
1626.33866183868
 20
296.7160046281916
 30
0.0
 10
1745.24459612016
 20
306.1720917982002
 30
0.0
 10
1824.510166696128
 20
311.90891354419
 30
0.0
 10
2062.292259321239
 20
327.5392634975415
 30
0.0
 10
2220.79415577079
 20
335.8526816976901
 30
0.0
 10
2696.245919829701
 20
355.4450722011516
 30
0.0
 10
3013.141808632271
 20
361.3762829349107
 30
0.0
 10
3883.754181435569
 20
365.2047369340217
 30
0.0
 10
4437.349959065938
 20
353.7779394056507
 30
0.0
 10
4990.839855556188
 20
342.3189730325903
 30
0.0
  0
ENDSEC
  0
EOF
