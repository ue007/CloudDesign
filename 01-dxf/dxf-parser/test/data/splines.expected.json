{
  "header": {},
  "entities": [
    {
      "type": "SPLINE",
      "handle": "3D",
      "ownerHandle": "157",
      "layer": "PLOTLINES",
      "normalVector": {
        "x": 0,
        "y": 0,
        "z": 1
      },
      "planar": true,
      "degreeOfSplineCurve": 3,
      "numberOfKnots": 9,
      "numberOfControlPoints": 5,
      "numberOfFitPoints": 3,
      "startTangent": {
        "x": 0.9514183274466854,
        "y": 0.3079012279913022,
        "z": 0
      },
      "knotValues": [
        0,
        0,
        0,
        0,
        1547.322808138777,
        5019.938770381674,
        5019.938770381674,
        5019.938770381674,
        5019.938770381674
      ],
      "controlPoints": [
        {
          "x": 0,
          "y": 0,
          "z": 0
        },
        {
          "x": 490.717092713168,
          "y": 158.8075309082932,
          "z": 0
        },
        {
          "x": 2164.905820751131,
          "y": 388.1223313710202,
          "z": 0
        },
        {
          "x": 3837.689038507039,
          "y": 353.4902462273104,
          "z": 0
        },
        {
          "x": 4994.861260930288,
          "y": 329.5329955163307,
          "z": 0
        }
      ],
      "fitPoints": [
        {
          "x": 0,
          "y": 0,
          "z": 0
        },
        {
          "x": 1522.67200320285,
          "y": 275.0960618560202,
          "z": 0
        },
        {
          "x": 4994.861260930288,
          "y": 329.5329955163307,
          "z": 0
        }
      ]
    },
    {
      "type": "SPLINE",
      "handle": "3E",
      "ownerHandle": "157",
      "layer": "PLOTLINES",
      "normalVector": {
        "x": 0,
        "y": 0,
        "z": 1
      },
      "planar": true,
      "degreeOfSplineCurve": 3,
      "numberOfKnots": 26,
      "numberOfControlPoints": 22,
      "numberOfFitPoints": 0,
      "knotValues": [
        0,
        0,
        0,
        0,
        294.0525150032295,
        294.0525150032295,
        1045.910856597887,
        1045.910856597887,
        1224.445418796694,
        1224.445418796694,
        1454.016505907184,
        1454.016505907184,
        1572.806679130005,
        1572.806679130005,
        1691.596852352826,
        1691.596852352826,
        1929.177198798468,
        1929.177198798468,
        2404.337891689753,
        2404.337891689753,
        3354.659277472323,
        3354.659277472323,
        5015.653135237717,
        5015.653135237717,
        5015.653135237717,
        5015.653135237717
      ],
      "controlPoints": [
        {
          "x": -3.910345595490071,
          "y": 12.08301275857048,
          "z": 0
        },
        {
          "x": 89.61730010503788,
          "y": 42.35074686835108,
          "z": 0
        },
        {
          "x": 184.0439740094698,
          "y": 69.42067550989128,
          "z": 0
        },
        {
          "x": 522.4770607652581,
          "y": 155.8817672278711,
          "z": 0
        },
        {
          "x": 770.4801029905291,
          "y": 199.9804308248713,
          "z": 0
        },
        {
          "x": 1079.587521541959,
          "y": 241.21623757007,
          "z": 0
        },
        {
          "x": 1139.003629790081,
          "y": 248.5213454592013,
          "z": 0
        },
        {
          "x": 1274.962277830939,
          "y": 264.0530430007511,
          "z": 0
        },
        {
          "x": 1351.545278854538,
          "y": 271.9460988128703,
          "z": 0
        },
        {
          "x": 1467.784975350118,
          "y": 282.9561993903517,
          "z": 0
        },
        {
          "x": 1507.426369672711,
          "y": 286.5407381210116,
          "z": 0
        },
        {
          "x": 1586.7020874704,
          "y": 293.4221664132911,
          "z": 0
        },
        {
          "x": 1626.33866183868,
          "y": 296.7160046281916,
          "z": 0
        },
        {
          "x": 1745.24459612016,
          "y": 306.1720917982002,
          "z": 0
        },
        {
          "x": 1824.510166696128,
          "y": 311.90891354419,
          "z": 0
        },
        {
          "x": 2062.292259321239,
          "y": 327.5392634975415,
          "z": 0
        },
        {
          "x": 2220.79415577079,
          "y": 335.8526816976901,
          "z": 0
        },
        {
          "x": 2696.245919829701,
          "y": 355.4450722011516,
          "z": 0
        },
        {
          "x": 3013.141808632271,
          "y": 361.3762829349107,
          "z": 0
        },
        {
          "x": 3883.754181435569,
          "y": 365.2047369340217,
          "z": 0
        },
        {
          "x": 4437.349959065938,
          "y": 353.7779394056507,
          "z": 0
        },
        {
          "x": 4990.839855556188,
          "y": 342.3189730325903,
          "z": 0
        }
      ]
    }
  ]
}